src/main/java/com/conveyal/analysis/results/BaseResultWriter.java

```diff
@@ -31,30 +30,20 @@ public abstract class BaseResultWriter {
 
     private static final Logger LOG = LoggerFactory.getLogger(BaseResultWriter.class);
 
+    public final String fileName;
     private final FileStorage fileStorage;
 
-    protected File bufferFile;
+    protected File bufferFile = FileUtils.createScratchFile();
 
-    public BaseResultWriter (FileStorage fileStorage) {
+    public BaseResultWriter(String fileName, FileStorage fileStorage) {
+        this.fileName = fileName;
         this.fileStorage = fileStorage;
     }
 
-    // Can this be merged into the constructor?
-    protected void prepare (String jobId) {
-        try {
-            bufferFile = File.createTempFile(jobId + "_", ".results");
-            // On unexpected server shutdown, these files should be deleted.
-            // We could attempt to recover from shutdowns but that will take a lot of changes and persisted data.
-            bufferFile.deleteOnExit();
```

Comment:
Instead of moving this logic to the constructor, this PR uses `FileUtils.createScratchFile()` at instantiation. That method does set `tempFile.deleteOnExit()`.

With this change, buffer file names will include a random component rather than  the jobId. That should be fine, I just wanted to note it for future debugging.